### ISSUE TITLE
Make the ZooKeeper config customizable in Accumulo Mini Cluster

### ISSUE THREAD

**Issue by BukrosSzabolcs** [apache/accumulo#1375]
Currently there is only a hardcoded list of zookeeper properties when MiniAccumuloCluster starts it's own zookeeper. It would be nice if we could define our custom Zookeeper config properties that would extend and if necessary override the hardcoded ones. Maybe something similar to the current siteConfig?

**Comment by joshelser** [comment]
Adding some more context onto this: Szabolcs was looking at this (along with a coworker, Maté) after a ZK 3.5 upgrade inside of Hive and the AccumuloStorageHandler.

ZK3.5 added an "AdminServer" (a little jetty server). The Accumulo-hive unit tests were failing because of a port-conflict where a local ZK was running and already bound on the port that the tests were trying to bind. It was a trivial fix to just configure ZK to not start the admin server.

It would be a good idea to expose API which lets us add/remove options as necessary.

**Comment by keith-turner** [comment]
Is the following the troublesome code?

https://github.com/apache/accumulo/blob/a60089890cff3a56dba2462b51971dcd6af69f48/minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java#L395

**Comment by keith-turner** [comment]
Would [MiniAccumuloConfig.setZooKeeperPort()](https://github.com/apache/accumulo/blob/a60089890cff3a56dba2462b51971dcd6af69f48/minicluster/src/main/java/org/apache/accumulo/minicluster/MiniAccumuloConfig.java#L94) help with this issue?  Not sure if that is the problematic port. If it is the javadoc says it will generate a random port if not set, I wonder if that is working properly.

**Comment by keith-turner** [comment]
To answer my own question, I don't think `MiniAccumuloConfig.setZooKeeperPort()` will help here, its a completely different port.

https://zookeeper.apache.org/doc/r3.5.5/zookeeperAdmin.html#sc_adminserver_config

**Comment by joshelser** [comment]
Yeah, the admin port is a completely different port than the server port.

**Comment by keith-turner** [comment]
Its a hack and I am not sure if it would work.  We could always set `admin.enableServer=false`.  If older versions of ZK ignore this property and newer versions don't start the admin server, that could be a way to fix this w/o any API changes.  This change could be made for 1.9.4 and 2.0.1
